src/crimson/osd/osd.cc

```diff
@@ -1265,21 +1265,21 @@ seastar::future<> OSD::prepare_to_stop()
 {
   if (osdmap && osdmap->is_up(whoami)) {
     state.set_prestop();
-    return monc->send_message(
+    (void) monc->send_message(
 	  make_message<MOSDMarkMeDown>(
 	    monc->get_fsid(),
 	    whoami,
 	    osdmap->get_addrs(whoami),
 	    osdmap->get_epoch(),
-	    true)).then([this] {
-      const auto timeout =
-	std::chrono::duration_cast<std::chrono::milliseconds>(
-	  std::chrono::duration<double>(
-	    local_conf().get_val<double>("osd_mon_shutdown_timeout")));
-      return seastar::with_timeout(
+	    true));
+    const auto timeout =
```

Comment:
thank you fixing this issue, @rzarzynski and i were wondering why the result of perf test was not stable, and we realized that `stop.sh` just failed to stop crimson-osd sometimes.

could you revise the commit message to expand the context a little bit, something like

> while performing test, there is chances that monitor is stopped before osd, in that case, osd would be waiting for the connection in vain for sending the `MOSDMarkMeDown`.

also, if we start ticking even before sending the message to monitor, i'd suggest guard all the stuff with `seastar::with_timeout()`, something like

```c++
const auto timeout =
        std::chrono::duration_cast<std::chrono::milliseconds>(
          std::chrono::duration<double>(
            local_conf().get_val<double>("osd_mon_shutdown_timeout")));

return seastar::with_timeout(seastar::timer<>::clock::now() + timeout,
  return monc->send_message(
          make_message<MOSDMarkMeDown>(
            monc->get_fsid(),
            whoami,
            osdmap->get_addrs(whoami),
            osdmap->get_epoch(),
            true)).then([this] {
  return stop_acked.get_future());
}).handle_exception_type([this](seastar::timed_out_error&) {
  return seastar::now();
});
```

sorry for the bikeshedding, i didn't realize that monitor could be offline when we send `MOSDMarkMeDown` to it when reviewing d2279d57ecd6d98e872b0bb35852f2fd4baa3f45 .

Reply:
All done:-)

> sorry for the bikeshedding, i didn't realize that monitor could be offline when we send `MOSDMarkMeDown` to it when reviewing [d2279d5](https://github.com/ceph/ceph/commit/d2279d57ecd6d98e872b0bb35852f2fd4baa3f45) .

Please don't be, I'm the one who should apologize for involving this mistake.